src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala

```diff
@@ -77,8 +80,15 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
     val put = new HttpPut(ps.bagIndexService.resolve(s"bags/$bagId").toASCIIString)
     logger.info(s"Adding new bag to bag index with request: ${ put.toString }")
     val response = http.execute(put)
-
-    if (response.getStatusLine.getStatusCode != HttpStatus.SC_CREATED) throw new IllegalStateException("Error trying to add bag to index")
+    val statusLine = response.getStatusLine
+    statusLine.getStatusCode match {
+      case HttpStatus.SC_CREATED =>
+      case HttpStatus.SC_BAD_GATEWAY =>
+        logger.error(s"Bad request while adding new bag to bag index  with message = ${ statusLine.getReasonPhrase }")
+        throw new IllegalStateException("Error trying to add bag to index")
+      case _ => //TODO shouldn't this also try to log as much as possible?
```

Comment:
yes, please! Since you discard the statusline at the end of this function (as you should), I would find it appropriate to have some logging here as well.